oneflow/core/functional/impl/math_functor.cpp

```diff
@@ -269,6 +269,78 @@ class TransposeFunctor {
   std::shared_ptr<OpExpr> op_;
 };
 
+class MovedimVecFunctor {
+ public:
+  MovedimVecFunctor() = default;
+  static Maybe<void> CheckNoRepeat(const std::vector<int32_t>& perm, std::vector<int32_t> &perm_out, int32_t indim, const std::string &desc) {  
+    std::vector<bool> is_used(indim, false);
+    FOR_RANGE(size_t, i, 0, perm.size()) {
+      int32_t item = perm[i];
+      if(item < 0){
+        item += indim;
+      }
+      CHECK_GE_OR_RETURN(item, 0) << ", Dimension out of range (expected to be in range of [" << -indim <<", " <<indim-1<< "], but got "<< perm[i] << ")";
+      CHECK_LT_OR_RETURN(item, indim) << ", Dimension out of range (expected to be in range of [" << -indim <<", " << indim-1 << "], but got " << perm[i] << ")";;
+      CHECK_EQ_OR_RETURN(is_used[item], false)<<"repeated dim in "<<desc;
+      
+      is_used[item] = true;
+      perm_out[i] = item;
+    }
+    return Maybe<void>::Ok();
+  }
+
+  Maybe<Tensor> operator()(const std::shared_ptr<one::Tensor>& input,
+                           const std::vector<int32_t>& source,
+                           const std::vector<int32_t>& destination ) const {
+    int32_t ndim = input->shape()->NumAxes();
+    int32_t dim = source.size();
+
+    CHECK_EQ_OR_RETURN(source.size(),destination.size())<< "movedim: Invalid source or destination dims: source (" <<
+              source.size() << " dims ) should contain the same number of dims as destination (" << destination.size() <<" dims)";
+
+    std::vector<int32_t> source_nopeat(dim);
+    std::vector<int32_t> destination_nopeat(dim);
+
+    CheckNoRepeat(source,source_nopeat,ndim,"source");
```

Comment:
```suggestion
    CheckNoRepeat(source, source_nopeat, ndim, "source");
```

这些是 `make of_format` 检查不出的吗？需要自己先 review 修正下